resources/js/components/shared/FilterGroup.vue

```diff
@@ -0,0 +1,199 @@
+            :type="result.typeInformation.inputFields.filter(f => f.type.kind !== 'LIST')[0].type.name"
+            :initial-field="filterToFilterRow(entry).field"
+            :initial-operator="filterToFilterRow(entry).operator"
+            :initial-value="filterToFilterRow(entry).value"
+            @delete="changeRow('deleted', index)"
+            @changeFilters="newEntry => changeRow(newEntry, index)"
+          />
+        </div>
+        <div class="tw-flex tw-flex-row tw-w-full tw-gap-1">
+          <button
+            class="tw-btn tw-btn-xs"
+            @click="addFilter"
+          >
+            <font-awesome-icon icon="fa-plus" /> Add Filter
+          </button>
+          <button
+            class="tw-btn tw-btn-xs"
+            @click="addGroup"
+          >
+            <font-awesome-icon icon="fa-bars-staggered" /> Add Group
+          </button>
+          <button
+            class="tw-btn tw-btn-xs"
+            @click="$emit('delete')"
+          >
+            <font-awesome-icon icon="fa-trash" /> Delete Group
+          </button>
+        </div>
+      </div>
+    </div>
+  </loading-indicator>
+</template>
+
+<script>
+import LoadingIndicator from './LoadingIndicator.vue';
+import {useQuery} from '@vue/apollo-composable';
+import gql from 'graphql-tag';
+import FilterRow from './FilterRow.vue';
+import {FontAwesomeIcon} from '@fortawesome/vue-fontawesome';
+
+export default {
+  components: { FontAwesomeIcon, FilterRow, LoadingIndicator },
+
+  props: {
+    type: {
+      type: String,
+      default: '',
+    },
+
+    /**
+     * A human-readable string to provide context for the plural type of record being filtered.
+     * This is expected to only be set for the top level of the recursive hierarchy.
+     *
+     * For example: "test measurements", "builds", "tests", etc...
+     */
+    primaryRecordName: {
+      type: String,
+      required: false,
+      default: '',
+    },
+
+    initialFilters: {
+      type: Object,
+      required: true,
+    },
+  },
+
+  emits: [
+    'changeFilters',
+    'delete',
+  ],
+
+  setup(props) {
+    const { result, error } = useQuery(gql`
+      query {
+        typeInformation: __type(name: "${props.type}") {
+          inputFields {
+            name
+            type {
+              name
+              kind
+              ofType {
+                name
+              }
+            }
+          }
+        }
+      }
+    `);
+
+    return {
+      result,
+      error,
+    };
+  },
+
+  data() {
+    return {
+      filters: JSON.parse(JSON.stringify(this.initialFilters)),
+    };
+  },
+
+  computed: {
+    currentCombineType() {
+      return this.filters.hasOwnProperty('any') ? 'any' : 'all';
+    },
+  },
+
+  methods: {
+    addGroup() {
+      this.filters[this.currentCombineType].push({
+        all: [
+          {},
+        ],
+      });
+    },
+
+    addFilter() {
+      this.filters[this.currentCombineType].push({});
```

Comment:
First pass of testing fails here for both `Add Filter` and `Add Group` on line 152: 
![image](https://github.com/user-attachments/assets/739b60f6-def1-4a13-a8f9-11e26b5c5ae6)

Reply:
Further info, this occurs when an empty filter is "applied" before any other interaction occurs.  With no filters created, I clicked on `Apply`.  The URL then looks like: `builds/16/tests?filters=[]`.  That means that we don't use the initial filter but the supplied one on all subsequent attempts to build a new filter which has no useful keys.